tap-agent/src/agent/sender_allocation.rs

```diff
@@ -599,8 +652,14 @@ impl SenderAllocationState {
             )
             .execute(&self.pgpool)
             .await
-            .map_err(|e| anyhow!("Failed to store failed receipt: {:?}", e))?;
+            .map_err(|e| anyhow!("Failed to store invalid receipt: {:?}", e))?;
         }
+        self.invalid_receipts_fees = self.calculate_invalid_receipts_fee().await?;
```

Comment:
Why doing `calculate_invalid_receipts_fee` here? It's already done in `pre_start`.

Reply:
This function is `store_invalid_receipts`, it's called when a successful rav has a few failed receipts.

We need to update the invalid receipt fees in this case, like what we do with unaggregated fees after every rav.

Reply:
Yeah but the reason we call `calculate_unaggregated_fee` when doing a rav request is that a successful rav request will delete a bunch of receipt from the DB. So we "need"* to resync with what's in the DB after that.

However there is no auto-deletion of receipts from the `scalar_tap_receipts_invalid` table anywhere, so there isn't a need to resync. It effectively can only grow, as pointed out here: https://github.com/graphprotocol/indexer-rs/pull/196/files/5e413c79dbca46edeef34645143938e324762563#diff-58ccff38a18daf85930e0a6a1a0a29557997648b983157e61c418df0f6a3cb1cR471. 
So just adding to `state.invalid_receipts_fees` after having loaded from the DB once in `pre_start` should be sufficient if I understand things correctly.

Notes:
\* well, we don't actually need to resync, we could instead subtract the successful RAV amount from unaggregated_fees. But resyncing makes sure we don't lose track of things accidentally.

Reply:
Do you think the suggestion is to add to the counter instead of recalculating it?

Reply:
Yup that was what I was thinking about.
However there may be something I don't see here, I may be wrong :shrug: